firestore/src/main/firestore_main.h

```diff
@@ -55,7 +56,8 @@ class Executor;
 class FirestoreInternal {
  public:
   // Note: call `set_firestore_public` immediately after construction.
-  explicit FirestoreInternal(App* app);
+  explicit FirestoreInternal(App* app,
+                             const char* database_id = kDefaultDatabase);
```

Comment:
We should strive to keep logic in one place. What I see is that the default value for `database_id` is being spread into many different files.

You could set default here, but then I think you should remove it from the Firestore class. But, since is both a `main` and `android` internal classes, then the logic is still duplicated. The least amount of duplicated code can be achieved if you set default in Firestore class, and then make it a required parameter on all layers below.